service/src/main/java/uk/gov/hmcts/reform/fpl/utils/JudgeAndLegalAdvisorHelper.java

```diff
@@ -53,6 +53,23 @@ public static void removeAllocatedJudgeProperties(JudgeAndLegalAdvisor judgeAndL
         judgeAndLegalAdvisor.setUseAllocatedJudge(null);
     }
 
+    public static JudgeAndLegalAdvisor prepareJudgeFields(JudgeAndLegalAdvisor judgeAndLegalAdvisor,
+                                                          Judge allocatedJudge) {
+        if (allocatedJudge.hasEqualJudgeFields(judgeAndLegalAdvisor)) {
+            return resetJudgeFields(judgeAndLegalAdvisor.getLegalAdvisorName());
+        } else {
+            judgeAndLegalAdvisor.setUseAllocatedJudge("No");
+            return judgeAndLegalAdvisor;
+        }
+    }
+
+    private static JudgeAndLegalAdvisor resetJudgeFields(String legalAdvisorName) {
+        return JudgeAndLegalAdvisor.builder()
+            .useAllocatedJudge("Yes")
+            .legalAdvisorName(legalAdvisorName)
+            .build();
+    }
```

Comment:
Does this make sense more sense to be on the JudgeAndLegalAdvisor object?

Reply:
I don't know :) What decides if it should be on the model or in a helper class?

Reply:
For now i've moved it to the JudgeAndLegalAdvisor object. It could equally live on both but I think it makes sense to live on the model.

Reply:
I moved it :)